### PR TITLE
Filter tests through Gradle properties

### PR DESCRIPTION
**Pull request by chkuang-g**
Enable test filtering through Gradle properties. See README.md for more information

In addition,    
- Update `reportAllTestsResult` task to include test type and test
  module information.
- Update `README.md` to include information about test-related
  properties.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -746,6 +746,74 @@ or Windows:
...
+
+* `INTERACTIVE_MODE_TESTS_ENABLED` - Default to `1`. Set to `1` to enable
+  interactive mode tests, which requires GPU on the machine. Otherwise, only run
+  tests in the batch mode.
+* `INCLUDE_TEST_TYPES` - Default to empty string, which means to include every
+  type of the test. To narrow down the types of test to run, set this properties
+  with a list of case-insensitive type strings separated by comma. For instance,
+  `-PINCLUDE_TEST_TYPES="Python,NUnit"` means to include only Python tests and
+  NUnit tests.
+  See `TestTypeEnum` in `build.gradle` for availble options.
```
Comment (a-maurice) [inline README.md:778]: available* (this happens a couple times in this)

File: README.md [hunk 2]
```diff
@@ -746,6 +746,74 @@ or Windows:
...
+* `EXCLUDE_TEST_TYPES` - Default to empty string, which means to exclude none.
+  To add types of testa to exclude, set this properties with
```
Comment (a-maurice) [inline README.md:780]: tests

**Review by a-maurice — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (chkuang-g) [reply README.md]: Good catch. Thank you.

File: README.md (hunk 2 above)
Comment (chkuang-g) [reply README.md]: Good catch. Thank you.

**Pull request merged**